no debug info by default in expect_test

trefis opened the pull request:
Context: I was adding some expect tests with "-dlambda -dno-unique-ids", in which I placed some empty expect blocks, and ran `make promote`.
At that point, the normal and "Principal" output differed, because:
- there were some debug infos in the lambda that I was dumping, which are printed with some location information.
- when expect_test is called for the second time on the file, with `-principal` all the locations past the first expect blocks are shifted by some amount.
So the `Principal {| ... |}` part is kept. Which means that if you run the tests once more, it will fail again, because the locations will have shifted some more.

The solution to this problem is to not generate debug info. Currently, the initialization of `Toploop` (which expect_tests links against) sets `Clflags.debug` to `true`.
Here I chose to unset it at the initialization of `Expect_test`. Another approach could be to not do the setting at the initialization of `Toploop` but instead do it in `Toploop.loop`, but I do not know if the current implementation was a conscious decision from @gasche or purely incidental. My guess is that we could probably move that setting to `Toploop.loop`, which seems cleaner than the approach proposed here, but I'll let @gasche decide that.

gasche commented:
I wondered why I was supposed to be an expert in Clflags placement, and the reason is that I was the one to commit 304c9c91a59b2092b68279b8a887a77de9ccc4c1 -- that was before the git era, I think.

I think that one would need to review what happens between initialization and the `loop` call to make sure that no code is produced (where `Clflags.debug` would matter), but moving the flag-set to `loop` sounds fine to me.

trefis committed:
add a test for MPR#6468

trefis committed:
toploop: set Clflags.debug to true from loop, not at toplevel

trefis commented:
>  that was before the git era, I think.

Ah, indeed!

Anyway, I also think it is fine to move the flag to loop:
- I added the example given in [MPR#6468](https://caml.inria.fr/mantis/view.php?id=6468) as a test
- removed the `Clflags.debug := true` line, and observed the following backtrace change:
  ```diff
   Exception: Not_found.
  -Raised at file "//toplevel//", line 5, characters 17-26
  -Called from file "//toplevel//", line 5, characters -15--11
  +Called from unknown location
   Called from file "toplevel/toploop.ml", line 180, characters 17-56
  ```
- added the line in `loop` and observed the initial backtrace reappearance.

gasche commented:
Do I correctly understand that the current 2-lines patch (plus tests) suffices to fix the issue you were after at the beginning?

trefis commented:
Yes.

trefis commented:
Thanks for the quick answers!

gasche commented:
(You merged before the CI results were in, hopefully you won't be caught again with a failing test :-)

trefis commented:
:>

I ran the testsuite locally this time, I'm 27% confident that everything will be fine.

trefis commented:
Apparently 27% was still too high; I'll submit a fix shortly.
